### DISCUSSION TITLE
WFDB and MIT meaning

### DISCUSSION THREAD

**Discussion by cx1111** [wfdb/wfdb-spec#3] [General]
We frequently use the term "WFDB Format" to refer to the MIT format signal and header files: https://www.physionet.org/physiotools/wag/signal-5.htm

From my understanding, it would be more accurate to describe such files as MIT format, WFDB compatible files. For instance, EDF files are also described as WFDB compatible.

So I'd like to clearly resolve a few questions:

- Should those file formats technically be called "MIT format", or "WFDB format"?
- What about annotation files?
- What is WFDB? A software package? What does it mean to be "WFDB compatible"?
- What are we actually trying to do? Document how the software package should behave, or document the file formats? There wouldn't be much point trying to re-document the WFDB software package, which the WPG/WAG already do.

**Comment by tompollard** [comment]
Great questions, I can't answer them but I've added a few comments below:

> Should those file formats technically be called "MIT format", or "WFDB format"?

In addition to this question, I think there are multiple, differing formats (format 8, format 16 etc) and this should be captured in the naming convention. Is there one "current" format, or do we choose different formats for different use cases?

> What is WFDB? A software package? What does it mean to be "WFDB compatible"?

It would be great to clear up the naming conventions. I think "WFDB" is short for **W**ave**F**orm **D**ata**B**ase, so presumably the name initially referred to a collection of files? Then the "WFDB library" came about, which is the "library of tools for reading, writing, and processing WFDB signals and annotations". 

> What are we actually trying to do? Document how the software package should behave, or document the file formats? There wouldn't be much point trying to re-document the WFDB software package, which the WPG/WAG already do.

Hmm, I think maybe we do need to re-document everything. The "WFDB Programmer’s Guide" is outdated and language specific. I think we need to try to work out exactly what WFDB means, create generalised documentation, and then add language-specific examples (either this or have non-language specific technical documents for format etc along with independently documented software packages).

**Reply by cx1111** [reply]
> In addition to this question, I think there are multiple, differing formats (format 8, format 16 etc) and this should be captured in the naming convention. Is there one "current" format, or do we choose different formats for different use cases?

The appropriate file format given the digital sampling resolution should be used: https://github.com/wfdb/wfdb-spec/pull/4

> Hmm, I think maybe we do need to re-document everything. The "WFDB Programmer’s Guide" is outdated and language specific. I think we need to try to work out exactly what WFDB means, create generalised documentation, and then add language-specific examples (either this or have non-language specific technical documents for format etc along with independently documented software packages).

So that's the thing. If WFDB means the original WFDB Software Package, then WAG/WPG will appropriately only apply to that package. Then we could just create a WFDB/MIT file spec.

**Comment by cx1111** [comment]
We've discussed the issue offline.

Changes in documentation to reflect this are made in: #5
